gate-web/src/main/groovy/com/netflix/spinnaker/gate/filters/RequestSheddingFilter.java

```diff
@@ -113,22 +115,34 @@ protected void shutdown() {
   }
 
   @Override
-  public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler)
-      throws Exception {
+  protected void doFilter(
+      HttpServletRequest request, HttpServletResponse response, FilterChain chain)
+      throws IOException, ServletException {
     if (configService.isEnabled(ENABLED_KEY, false) && isDroppable(request)) {
       if (shouldDropRequestWithChance()) {
         log.warn("Dropping low priority request: {}", request.getRequestURI());
         registry.counter(requestsId.withTag("action", "dropped")).increment();
 
         response.setDateHeader(
             "Retry-After", Instant.now().plus(Duration.ofMinutes(1)).toEpochMilli());
-        throw new LowPriorityRequestRejected();
+
+        // Raising an exception here (or using response.sendError) would result in the request being
+        // dispatched to `/error` and triggering an evaluation of the spring security filter chain.
+        //
+        // This filter is attempting to shed requests _prior_ to the spring filter chain being
+        // evaluated and thus needs to more explicit around how the status and response body
+        // is constructed.
+        response.setStatus(SERVICE_UNAVAILABLE.value());
```

Comment:
`response.sendError()` or raising an exception here triggers the spring security filter chain to be evaluated due to some behind the scenes behavior around the request being redirected to `/error`. 

I couldn't find a way to avoid the security filter short of setting `spring.security.filter.dispatcher-types: async,request` which is maybe heavy handed? 

Semi-related links:
* https://stackoverflow.com/questions/37138529/does-the-spring-security-filterchain-always-run-twice-for-asynchronous-requests
* https://github.com/spring-projects/spring-security/issues/5359